clang/lib/AST/OpenACCClause.cpp

```diff
@@ -72,3 +92,9 @@ void OpenACCClausePrinter::VisitDefaultClause(const OpenACCDefaultClause &C) {
 void OpenACCClausePrinter::VisitIfClause(const OpenACCIfClause &C) {
   OS << "if(" << C.getConditionExpr() << ")";
 }
+
+void OpenACCClausePrinter::VisitSelfClause(const OpenACCSelfClause &C) {
+  OS << "self";
+  if (const Expr *CondExpr = C.getConditionExpr())
```

Comment:
Need to check if Clause.hasConditionExpr() is true?

Reply:
No, all that does is check if `CondExpr` is `nullptr`, so this is doing the same thing.  I thought this interface makes sense (that is, 'null' means no condition).